cmd/cli/root.go

```diff
@@ -85,13 +76,24 @@ func NewRootCmd() *cobra.Command {
 			ctx = context.WithValue(ctx, spanKey, span)
 
 			cmd.SetContext(ctx)
+
 		},
 		PersistentPostRun: func(cmd *cobra.Command, args []string) {
 			ctx := cmd.Context()
 			ctx.Value(spanKey).(trace.Span).End()
 			ctx.Value(util.SystemManagerKey).(*system.CleanupManager).Cleanup(ctx)
 		},
 	}
+	defaultRepo, err := defaultRepo()
+	if err != nil {
+		panic(err)
+	}
+	RootCmd.PersistentFlags().String("repo", defaultRepo, "path to bacalhau repo")
+
+	if err := configflags.RegisterFlags(RootCmd, rootFlags); err != nil {
+		panic(err)
+	}
+
```

Comment:
I've added a `--repo` flag to the root command that can be used to configure where bacalhau creates its repo, if left blank the default `$HOME/.bacalhau` is used. The existing env vars are also still respected.